test/color/test_gray_via_conv.py

```diff
@@ -0,0 +1,154 @@
+import pytest
+import torch
+from torch.autograd import gradcheck
+
+import kornia.color as color
+import kornia.color.via_conv as color_via_conv
+from kornia.testing import BaseTester, assert_close
+
+
+class TestGrayscaleToRgb(BaseTester):
+    def test_smoke(self, device, dtype):
+        B, C, H, W = 2, 1, 4, 5
+        img = torch.rand(B, C, H, W, device=device, dtype=dtype)
+        assert isinstance(color_via_conv.grayscale_to_rgb(img), torch.Tensor)
+
+    @pytest.mark.parametrize("batch_size, height, width", [(1, 3, 4), (2, 2, 4), (3, 4, 1)])
+    def test_cardinality(self, device, dtype, batch_size, height, width):
+        img = torch.ones(batch_size, 1, height, width, device=device, dtype=dtype)
+        assert color_via_conv.grayscale_to_rgb(img).shape == (batch_size, 3, height, width)
+
+    def test_exception(self, device, dtype):
+        with pytest.raises(TypeError):
+            assert color_via_conv.grayscale_to_rgb([0.0])
+
+        with pytest.raises(TypeError):
+            img = torch.ones(1, 1, device=device, dtype=dtype)
+            assert color_via_conv.grayscale_to_rgb(img)
+
+        with pytest.raises(TypeError):
+            img = torch.ones(2, 1, 1, device=device, dtype=dtype)
+            assert color_via_conv.grayscale_to_rgb(img)
+
+        with pytest.raises(TypeError):
+            img = torch.ones(1, 3, 1, 1, device=device, dtype=dtype)
+            assert color_via_conv.grayscale_to_rgb(img)
+
+    def test_color(self, device, dtype):
+        data = torch.rand(2, 1, 5, 5, device=device, dtype=dtype)
+        assert_close(color.grayscale_to_rgb(data), color_via_conv.grayscale_to_rgb(data))
```

Comment:
It's better to compare against a hard-coded case (can be done in the `test_smoke` instead of this test). Same for the others tests